prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -283,6 +291,24 @@ class ReportFunction(
         return responseBuilder.build()
     }
 
+    fun responseContainsQualityFilter(response: String): Boolean {
+        var hasQualityFilters = false
+        val tree = jacksonObjectMapper().readTree(response)
```

Comment:
Rather than undoing the jackson mapping use the tracked `ActionLogs` within the `actionResponse`, you can filter by type (in this case `fitler` or similar) and simply check to see if any exist.

Reply:
@whytheplatypus this method was rewritten and unit tests add. please re-review the whole spiel. thanks for your help!